contracts/ExampleToken.cdc

```diff
@@ -106,10 +88,66 @@ pub contract ExampleToken: FungibleToken {
                 ExampleToken.totalSupply = ExampleToken.totalSupply - self.balance
             }
         }
+
+        /// The way of getting all the Metadata Views implemented by ExampleToken
+        ///
+        /// @return An array of Types defining the implemented views. This value will be used by
+        ///         developers to know which parameter to pass to the resolveView() method.
+        ///
+        pub fun getViews(): [Type]{
+            return [Type<FungibleTokenMetadataViews.FTView>(),
+                    Type<FungibleTokenMetadataViews.FTDisplay>(),
+                    Type<FungibleTokenMetadataViews.FTVaultData>()]
+        }
+
+        /// The way of getting a Metadata View out of the ExampleToken
+        ///
+        /// @param view: The Type of the desired view.
+        /// @return A structure representing the requested view.
+        ///
+        pub fun resolveView(_ view: Type): AnyStruct? {
+            switch view {
+                case Type<FungibleTokenMetadataViews.FTView>():
+                    return FungibleTokenMetadataViews.FTView(
+                        ftDisplay: self.resolveView(Type<FungibleTokenMetadataViews.FTDisplay>()) as! FungibleTokenMetadataViews.FTDisplay?,
+                        ftVaultData: self.resolveView(Type<FungibleTokenMetadataViews.FTVaultData>()) as! FungibleTokenMetadataViews.FTVaultData?
+                    )
```

Comment:
I find it easier to read the `resolveView` function when it uses helpers for each view type.

```cdc
pub fun resolveView(_ view: Type): AnyStruct? {
  switch view {
    case Type<FungibleTokenMetadataViews.FTView>():
      return self.resolveFTView()
    case Type<FungibleTokenMetadataViews.FTDisplay>():
      return self.resolveFTDisplayView()
      
    // ...
  }
}

// You can then avoid those clunky cast operations when resolving sub-views
pub fun resolveFTView(): FungibleTokenMetadataViews.FTView {
  return FungibleTokenMetadataViews.FTView(
    // Note: Cadence will automatically coerce FungibleTokenMetadataViews.FTView
    // into the optional FungibleTokenMetadataViews.FTView?
    ftDisplay: self.resolveFTDisplayView(),
    ftVaultData: self.resolveFTVaultDataView()
  )
}

// ...
```